doc/Getting_Started.md

```diff
@@ -91,76 +97,153 @@ directory.  All boards share a few common targets:
+The process to load the kernel onto the board depends on the board.
+There are two main variants: some boards (notably the
+[Imix](../boards/imix/README.md) and [Hail](../boards/hail/README.md)
+boards) have a serial bootloader, most other boards use a programming
+adapter that supports the JTAG or SWD protocol instead.
 
-- [libtock-c](https://github.com/tock/libtock-c): C and C++ apps.
-- [libtock-rs](https://github.com/tock/libtock-rs): Rust apps.
+To load a kernel onto a board using a serial bootloader, no other
+software is required and you can just run
 
-Compiled applications are architecture-specific (e.g. `cortex-m4`,
-`cortex-m0`) since the compiler emits slightly different instructions
-for each variant. Compiled applications can also depend on specific
-drivers, which not all boards provide; if you load an application onto
-a board that does not support every driver/system call it uses, some
-system calls with return error codes (`ENODEVICE` or `ENOSUPPORT`).
+    $ make program
 
-Applications are built for all architectures Tock supports. Boards select an
-appropriate architecture when uploading code (e.g. `cortex-m4` for the SAM4L on
-the `imix` board). Apps are packaged into .tab files that contain compiled
-binaries for all supported architectures.
+in the board's directory. To load the kernel using a programming
+adapter, you need the appropriate software that supports the adapter
+and can then install the kernel by running
 
-## Loading the kernel and applications onto a board
+    $ make flash
 
-To load a kernel onto a board using a serial bootloader, run
+Depending on the apapter, you will need either the free `openocd` or
+Segger's proprietary `JLinkExe`. Programming adapters are available as
+standalone devices (for example the [JLink EDU JTAG
+debugger](https://www.segger.com/j-link-edu.html) available on
+[Digikey](https://www.digikey.com/product-detail/en/segger-microcontroller-systems/8.08.90-J-LINK-EDU/899-1008-ND/2263130)),
+but most development boards come with an onboard programming and
+debugging adapter. In that case, the board you use determines which
+software you will need and the `Makefile` in the board directory will
+know which one to call. Again, the [board-specific
+READMEs](../boards/README.md) provide the required details.
 
-    $ make program
+### Installing `JLinkExe`
 
-in the board's directory. To load the kernel using JTAG, run
+`JLink` is available [from the Segger
+website](https://www.segger.com/downloads/jlink). You want to install
+the "J-Link Software and Documentation Pack". There are various
+packages available depending on operating system. We require a version
+greater than or equal to `5.0`.
 
-    $ make flash
+### Installing `openocd`
+
+`Openocd` works with various programming and debugging adapters. For
+most purposes, available distribution packages are sufficient and it can
+be installed with:
+
+```bash
+(Linux): sudo apt-get install openocd
```

Comment:
This applies only to Debian based distros, not all Linux distros.

Reply:
It has always been the case that "Linux" in the TockOS documentation means "Ubuntu", I haven't changed that. Should I? And if so, how?

Reply:
That seems very confusing. I would just write (Linux/Debian) in the bracket so it's clear